veggieseasons/lib/screens/search.dart

```diff
@@ -12,19 +12,25 @@ import 'package:veggieseasons/widgets/search_bar.dart';
 import 'package:veggieseasons/widgets/veggie_headline.dart';
 
 class SearchScreen extends StatefulWidget {
+  SearchScreen({this.restorationId, Key key}) : super(key: key);
+
+  final RestorationId restorationId;
+
   @override
   _SearchScreenState createState() => _SearchScreenState();
 }
 
-class _SearchScreenState extends State<SearchScreen> {
-  final controller = TextEditingController();
+class _SearchScreenState extends State<SearchScreen> with RestorationMixin {
+  RestorableTextEditingController controller = RestorableTextEditingController(TextEditingController());
   final focusNode = FocusNode();
-  String terms = '';
+  String terms;
 
   @override
   void initState() {
     super.initState();
+    register(controller, const RestorationId('text'));
     controller.addListener(_onTextChanged);
+    terms = controller.value.text;
```

Comment:
Would it be possible to just route the properties directly inside the `RestorableTextEditingController` class:

```
String get text => _value.text;
```

in order to make `terms = controller.value.text` back into `terms = controller.text`?

Reply:
Interesting idea. I think that may be possible. Let me experiment with that...

Reply:
I'm guessing this didn't end up being feasible/advisable? 😄

Reply:
As mentioned in the comment about AppState, merging the two also makes it harder to throw good/proper error messages when you forget to register the class.